NiBAx/plugins/harmonization/harmonization.py

```diff
@@ -77,13 +78,13 @@ def OnLoadHarmonizationModelBtnClicked(self):
             self.ui.Harmonized_Data_Information_Lbl.setObjectName('Missing_label')
             self.ui.Harmonized_Data_Information_Lbl.setStyleSheet('QLabel#Missing_label {color: red}')
         else:
-            self.datamodel.harmonization_model = pd.read_pickle(filename)
-            if not (isinstance(self.datamodel.harmonization_model,dict) and 'SITE_labels' in self.datamodel.harmonization_model):
-                text_2=('Selected file is not a viable harmonization model')
-                self.ui.Harmonized_Data_Information_Lbl.setText(text_2)
-                self.ui.Harmonized_Data_Information_Lbl.setObjectName('Error_label')
-                self.ui.Harmonized_Data_Information_Lbl.setStyleSheet('QLabel#Error_label {color: red}')
+            harmonization_model = pd.read_pickle(filename)
+            if not self.datamodel.IsValidHarmonization(harmonization_model):
+                self.datamodel.SetHarmonizationModelFilePath(None)
+                self.datamodel.SetHarmonizationModel(None)
```

Comment:
Do we really need these lines here? This will emit the signal 'harmonization_model_changed' unnecessarily from datamodel when we set 'None' and may lead to unexpected behavior.

Reply:
I think we need it, because the harmonization model is used for instance in age trends to draw the confidence intervals. If the harmonization model changes, the plot has to change as well.